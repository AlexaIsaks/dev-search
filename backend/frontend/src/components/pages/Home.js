import React from "react";
import SearchResult from "../components/SearchResult";

// Home page
const Home = ({state, handleSearchChange, handleSearchSubmit, handleDisplayUser,}) => {
  const {searchResults, isSearching, isLoaded, error } = state;
  let displaySearchResults = null;

  // An error occurred during the search
  if (error) {
    displaySearchResults = (<p>An error has occurred.</p>);
  // Search is not in process
  } else if (!isSearching) {
    displaySearchResults = null;
  // An empty array is returned, thus no users were found
  } else if (isLoaded && searchResults.length === 0) {
    displaySearchResults = (<p>No results. Please ensure that you have entered the correct username.</p>);
  // User(s) found
  } else if (isLoaded && searchResults.length > 0) {
    displaySearchResults = searchResults.map((result) => {
      return (
        <SearchResult
          key={result.id}
          profile={result}
          handleDisplayUser={handleDisplayUser}
        />
      );
    });
  // Waiting on search results
  } else {
    displaySearchResults = <p>Please wait...</p>;
  }

  return (
    <React.Fragment>
      
      {/*Page header*/}
      <header className="py-5">
        <h1 className="h2 text-center text-uppercase mb-0">Dev Search</h1>
      </header>

      {/*Page main*/}
      <main>
        <div className="container px-md-5">
          <div>

            {/*Search form */}
            <form onSubmit={handleSearchSubmit} className="w-100">
              <div className="input-group">
                <input
                  type="search"
                  name="search"
                  placeholder="Search dev by username"
                  onChange={handleSearchChange}
                  className="form-control border-top-0 border-start-0 border-end-0 bg-offwhite input"
                  required
                />
                <button
                  type="submit"
                  className="btn border border-top-0 border-start-0 border-end-0"
                >
                  <span className="d-flex justify-content-center align-items-center fs-5 text-mustard">
                    <ion-icon name="search-outline"></ion-icon>
                  </span>
                </button>
              </div>
            </form>
          </div>

          {/*Search results */}
          <div className="py-5">
            {displaySearchResults}
          </div>
        </div>
      </main>
    </React.Fragment>
  );
};

export default Home;
