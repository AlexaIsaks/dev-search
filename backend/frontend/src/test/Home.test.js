import React from "react";
import Home from "../components/pages/Home";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

// Home component snapshot test
test("Home component renders correctly", () => {
  // Mock state: Search results
  const state = {
    error: null,
    isLoaded: true,
    isSearching: true,
    search: "john_smith",
    searchResults: [
      {
        avatar:
          "https://secure.gravatar.com/avatar/2ed3e45ee10bf4e18df94e547c891c91?s=80&d=identicon",
        bio: "",
        id: 2705266,
        name: "John Smith ",
        platform: "GitLab",
        username: "John_smith",
        webUrl: "https://gitlab.com/John_smith",
      },
    ],
    userProfile: null
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <Home state={state} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
