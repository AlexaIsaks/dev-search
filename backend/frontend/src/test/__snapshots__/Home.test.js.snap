// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
Array [
  <header
    className="py-5"
  >
    <h1
      className="h2 text-center text-uppercase mb-0"
    >
      Dev Search
    </h1>
  </header>,
  <main>
    <div
      className="container px-md-5"
    >
      <div>
        <form
          className="w-100"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control border-top-0 border-start-0 border-end-0 bg-offwhite input"
              name="search"
              placeholder="Search dev by username"
              required={true}
              type="search"
            />
            <button
              className="btn border border-top-0 border-start-0 border-end-0"
              type="submit"
            >
              <span
                className="d-flex justify-content-center align-items-center fs-5 text-mustard"
              >
                <ion-icon
                  name="search-outline"
                />
              </span>
            </button>
          </div>
        </form>
      </div>
      <div
        className="py-5"
      >
        <div
          className="mb-3 border-bottom bg-white shadow result"
        >
          <a
            className="mb-3 text-secondary text-decoration-none"
            data-id={2705266}
            href="/user-details/2705266"
            onClick={[Function]}
          >
            <div
              className="row gx-2 gx-md-3 h-100"
            >
              <div
                className="col-4 col-md-2 h-100 result__image-container"
              >
                <img
                  alt="John Smith "
                  className="result__image"
                  src="https://secure.gravatar.com/avatar/2ed3e45ee10bf4e18df94e547c891c91?s=80&d=identicon"
                />
              </div>
              <div
                className="col-6 col-md-9 d-flex flex-column justify-content-center align-items-start"
              >
                <span>
                  John Smith 
                   (
                  John_smith
                  )
                </span>
                <span>
                  GitLab
                </span>
              </div>
              <div
                className="col-2 col-md-1 d-flex justify-content-center align-items-center"
              >
                <span
                  className="d-flex justify-content-center align-items-center fs-5 text-mustard"
                >
                  <ion-icon
                    name="chevron-forward-outline"
                  />
                </span>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </main>,
]
`;
